src/embeddings/embeddings_api.hpp

```diff
@@ -0,0 +1,34 @@
+//*****************************************************************************
+// Copyright 2024 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#pragma once
+
+#include <string>
+#include <variant>
+#include <vector>
+
+#include "rapidjson/document.h"
+
+enum class EncodingFormat {
+    FLOAT,
+    BASE64
+};
+
+struct EmbeddingsRequest {
+    std::variant<std::vector<std::string>, std::vector<std::vector<int>>> input;
+    EncodingFormat encoding_format;
```

Comment:
L31/L33 please follow contributing guide https://github.com/openvinotoolkit/model_server/blob/0cef8c9702b49b8ba42bdef1ab63d6c4f141f1c4/docs/contributing.md?plain=1#L31

Reply:
According to https://github.com/openvinotoolkit/model_server/blob/0cef8c9702b49b8ba42bdef1ab63d6c4f141f1c4/docs/contributing.md?plain=1#L32 L31 should be fine? And this would be consistent with field name in request json.